src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -238,7 +238,6 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
                 pattern = BindPattern(node.Pattern, sliceType, GetValEscape(sliceType, inputValEscape), permitDesignations, hasErrors, diagnostics);
             }
 
-            Debug.Assert(GetIndexerOrImplicitIndexerSymbol(indexerAccess) is var _);
```

Comment:
>GetIndexerOrImplicitIndexerSymbol [](http://example.com/codeflow?start=25&length=33)

I think we should keep this assert, unless we are removing this method or changing how it is used.